oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java

```diff
@@ -326,6 +327,38 @@ public void refreshAccessToken_withoutServiceAccountImpersonation() throws IOExc
     assertEquals(transportFactory.transport.getAccessToken(), accessToken.getTokenValue());
   }
 
+  @Test
```

Comment:
A test with clientId and userProject validating that userProject gets priority? With separate credential this can be a ExternalCredential test.

Reply:
Client ID/secret are not currently used.